pages/_app/index.jsx

```diff
@@ -41,22 +55,29 @@ class App extends NextApp {
     this.props.router.push(route.href, route.as)
   }
 
-  render() {
-    const { Component, pageProps, router } = this.props
-    const { store } = this
-
-    if (store === null) return null
+  static getDerivedStateFromProps(props, state) {
+    const { router } = props
+    const { store } = state
 
+    if (!store) return null
```

Comment:
I am not sure, you can have `state == null`. You may find destructuring issues.

Reply:
When it can happen?

Reply:
It can happen when you blindly start destructuring state in `render()` like `const { store } = this.state` what you actually do. If you try `const store = null.store` you will face the problem.

Reply:
I mean when `state` can be null?

Reply:
As far as I understand the method returns new state based on current props and current state. If you return null you may get state being null. I am not sure now.

Reply:
`getDerivedStateFromProps is invoked right before calling the render method, both on the initial mount and on subsequent updates. It should return an object to update the state, or null to update nothing.`

https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

Reply:
Thank you for the explanation 👍